src/test/groovy/com/jvm_bloggers/core/rss/BlogPostsControllerSpec.groovy

```diff
@@ -0,0 +1,114 @@
+package com.jvm_bloggers.core.rss
+
+import com.jvm_bloggers.core.rss.json.RssToJsonConverter
+import com.rometools.rome.feed.synd.*
+import org.apache.commons.io.IOUtils
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+import java.nio.charset.Charset
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup
+
+class BlogPostsControllerSpec extends Specification {
+
+    Date today = new Date(0)
+
+    AggregatedRssFeedProducer rssProducerMock = Mock() {
+        SyndLink link = new SyndLinkImpl()
+        link.rel = "self"
+        link.href = "http://jvm-bloggers.com/rss"
+
+        SyndFeed feed = new SyndFeedImpl()
+        feed.links = [link]
+        feed.uri = "URI"
+        feed.feedType = AggregatedRssFeedProducer.FEED_TYPE
+        feed.title = AggregatedRssFeedProducer.FEED_TITLE
+        feed.description = AggregatedRssFeedProducer.FEED_DESCRIPTION
+        feed.publishedDate = today
+        feed.entries = [createSyndEntry("postId", "postUrl", "postTitle", "postAuthor", "postDescription", today)]
+
+        1 * getRss(_, _, _) >> feed
+    }
+
+    RssToJsonConverter rssToJsonConverter = new RssToJsonConverter("http://jvm-bloggers.com")
+
+    @Subject
+    BlogPostsController blogPostsController = new BlogPostsController(rssProducerMock, rssToJsonConverter, 50)
+
+    @Unroll
+    def "Should get OK status for RSS feed in #format format request"() {
+        given:
+            MockMvc mockMvc = standaloneSetup(blogPostsController).build()
```

Comment:
No indentation after labels, convention was simplified to

    given:
    something;
    
    when:
    something;

    then:
    something;

Docs should be updated, but if you have found it somewhere, blame is on me :)